other/relax/relax_driver.jl

```diff
@@ -0,0 +1,21 @@
+using Flux, Statistics
+using Flux.Optimise: update!
+include("relax.jl")
+
+D = 100
+θ = param(zeros(D))
+c = collect(range(0,1,length = D))
+f(x) = sum((x .- c).^2, dims = 1)
+b = BernoulliRelax(f, Chain(Dense(D, 5, relu), Dense(5, 1)))
+
+ϕ = params(b)
+push!(ϕ, θ)
+nsamples = 10
+
+opt = ADAM(0.1)
+@elapsed for i in 1:2000
+    fVal = ∇relax!(θ, b, nsamples)
+    update!(opt, ϕ)   
```

Comment:
The confusion above manifests itself here. It looks like you're doing some gradient of relax and it only takes in θ yet in the line below, `update!(opt, ϕ) ` so clearly something happened to the ϕ gradients but it's not obvious.

What I meant before by having the same interface is that the goal is that there should be two lines that look like:
```
update!(opt, θ)
update!(opt, ϕ)
```

Reply:
I do not know, how to really resolve this. May-be, to overload api of Flux.
Something like
```
Flux.gradient(relax,...)
```
Other option I am thinking about would be to try to seamlessly include it into the Flux TrackedArray.

Frankly speaking, I do not know, how to do this at the moment.